WordPress/gutenberg
# [RNMobile] - Update tests with paragraph blocks to work on both environments and platforms

## What?
This is a follow up to https://github.com/WordPress/gutenberg/pull/40078. However, instead of just updating `gutenberg-editor-paragraph.test.js`, this PR updates all tests that are using the paragraph block. The reason is because adding a condition to make it only work on the `gutenberg-editor-paragraph.test.js` file seems to be more complicated as the functions are used on multiple tests. So in the end, decided that it made more sense to update it this way. This PR does not fix tests that are not using paragraph blocks. That will be handled separately in a separate PR(s). 

## Why?

This PR will attempt to fix the flakiness that we see on tests related to the paragraph block. This PR will also make the tests work in both local and CI environments. Previously, the tests only work on CI with some flakiness every once in a while.

## How?

- Created a new function `getTextBlockLocatorAtPosition()` for paragraph block which will now wait for the block to be visible and return the locator for the block at the chosen position.
- Replaced one of the validations that check for paragraph blocks separately into this `getNumberOfParagraphBlocks()` that will check the number of paragraph blocks that exist on the screen.
- Remove all implicit waits in updated tests as we don't need them anymore after using `waitForVisible()`

## Testing Instructions
Ensure that the affected test files work locally and in CI.

For local testing, use the same simulator/emulator that CI is using:
iOS - iPhone 11, iOS 14.5
Android - Google Pixel 3 XL GoogleAPI Emulator, Android 9.0

This was also tested [here](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile?branch=version-toolkit%2Fgutenberg%2Frnmobile%2Ffix-tests-using-paragraph-blocks&filter=all) and all updated tests are passing 6 out of 6 times. There are a few failing test cases in the test runs that will be handled separately. Failing tests from the test runs:
1. Android - Gutenberg Editor tests for List block (end) should be able to end a List block 
2. iOS - Gutenberg Editor Spacer Block test should be able to add a spacer block
3. One build failed because [Sauce Labs did not start](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/17070/workflows/a4135300-ad85-4e8b-b4bc-4362bc041f3c/jobs/92851?invite=true#step-119-393)
## Screenshots or screencast
